src/transformers/models/mllama/modeling_mllama.py

```diff
@@ -1972,16 +1970,11 @@ def forward(
         slice_indices = slice(-logits_to_keep, None) if isinstance(logits_to_keep, int) else logits_to_keep
         logits = self.lm_head(hidden_states[:, slice_indices, :]).float()
 
-        loss = None
-        if labels is not None:
-            loss = self.loss_function(logits, labels, self.vocab_size, **loss_kwargs)
-
         if not return_dict:
             output = (logits,) + outputs[1:]
-            return (loss,) + output if loss is not None else output
+            return output
 
         return CausalLMOutputWithPast(
-            loss=loss,
```

Comment:
I still left rest of return types in the class , for minimal changes as you mentioned there would be a massive refactor @zucchini-nlp . So I just moved loss computation to ConditionalGenerationClass . However with that change, there may not be a need for this `MllamaForCausalLM` as basically its just adding logits, which can also be moved and just the MllamaText Model class can be used directly from wrapper.

Will leave that to you as you think of the refactor. Or if you want me to clean it up, can do

Reply:
I will add it back for users that use this class directly, but will not pass labels from GenerationClass to avoid exception